### PR TITLE
SQL-579: Refactor repeated new calls to class-level constants in MongoSQLDatabaseMetadata

### PR DESCRIPTION
**Pull request by kli19**
Replace the following repeated calls with the corresponding class-level constants
- `new BsonInt32(0)` -> `BSON_ZERO_INT_VALUE`
- `new BsonInt32(typeSearchable)` -> `BSON_TYPE_SEARCHABLE_INT_VALUE`
- `new BsonInt32(Types.OTHER)` -> `BSON_OTHER_INT_VALUE`
- `new BsonInt32(ResultSetMetaData.columnNullable)` -> `BSON_COLUMN_NULLABLE_INT_VALUE `
- `new BsonBoolean(false)` -> `BsonBoolean.FALSE`
- `new BsonNull()` -> `BsonNull.VALUE`
- `new BsonString("")` -> `BSON_EMPTY_STR_VALUE`
- `new BsonString("YES")` -> `BSON_YES_STR_VALUE`
- `new BsonString("NO")` -> `BSON_NO_STR_VALUE`

### CODE REVIEW

**Comment by kli19** [comment]
evergreen merge

File: src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java
Comment (bucaojit) [inline src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java:1154]: We can also refactor these `new BsonInt32(typeSearchable))` calls as typeSearchable is a final int value `public static final int typeSearchable = 3`.

File: src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java
Comment (bucaojit) [inline src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java:1128]: `new BsonInt32(ResultSetMetaData.columnNullable)` is another good candidate.

File: src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java
Comment (mattChiaravalloti) [inline src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java:1125]: I noticed that other places in the file, we use `BsonNull.VALUE`, which is a class-level constant defined _in_ the java driver. They also have `BsonBoolean.FALSE`. Maybe we should defer to those constants instead of creating our own `BSON_NULL_VALUE` and `BSON_FALSE_VALUE` constants in this class! That we we're just making things like `BSON_ZERO_VALUE` that don't already exist somewhere.

File: src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java
Comment (mattChiaravalloti) [inline src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java:56]:
```suggestion
    private static final BsonString BSON_EMPTY_STR_VALUE = new BsonString("");
```
or
```
    private static final BsonString BSON_EMPTY_STRING_VALUE = new BsonString("");
```
[nit] I think the name "BSON_EMPTY_VALUE" is a bit misleading! Knowing the type helps a lot, but when reading through the code below, there isn't any visual indication of the type. I know it is a bit of an antipattern to include type info in variable names, but in this case it will help disambiguate what it represents. (For example, I could also name the empty array _or_ empty document constant `BSON_EMPTY_VALUE`, and there would be no way of knowing which type I'm referring to without going back to the declaration!)

**Review by mattChiaravalloti — CHANGES_REQUESTED** [review]
Well done so far 🥇 I just have a couple more notes on top of what Oliver already suggested!

**Comment by kli19** [comment]
evergreen merge

**Comment by kli19** [comment]
evergreen merge

**Review by bucaojit — APPROVED** [review]
LGTM!
Thanks for making the suggested changes.

After one more approval, address any possible last-minute changes and confirm the evergreen tests pass.  Then you can leave a comment of `evergreen merge` and evergreen will run tests and merge the code for you.

**Review by mattChiaravalloti — APPROVED** [review]
lgtm 🎉

**Comment by kli19** [comment]
evergreen merge

**Pull request merged**
